styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java

```diff
@@ -0,0 +1,295 @@
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    flyteRunner = new FlyteAdminClientRunner(flyteAdminClient, stateManager);
+  }
+
+  @Test
+  public void testIsEnabled() {
+    assertTrue(flyteRunner.isEnabled());
+  }
+
+  @Test
+  public void testCreateExecution() throws FlyteRunner.CreateExecutionException {
+    when(flyteAdminClient.createExecution(any(), any(), any(), any(),any())).thenReturn(
+        ExecutionOuterClass.ExecutionCreateResponse
+            .newBuilder()
+            .setId(IdentifierOuterClass.WorkflowExecutionIdentifier
+                .newBuilder()
+                .setProject("flyte-test")
+                .setDomain("testing")
+                .setName("test-create-execution")
+                .build())
+            .build());
+    var flyteExecution = flyteRunner.createExecution("test-create-execution", FLYTE_EXEC_CONF);
+
+    assertThat(flyteExecution.getProject(), is("flyte-test"));
+    assertThat(flyteExecution.getDomain(), is("testing"));
+    assertThat(flyteExecution.getName(), is("test-create-execution"));
+    assertThat(flyteExecution.toUrn(), is("ex:flyte-test:testing:test-create-execution"));
+  }
+
+  @Test
+  public void testThrowsFlyteLaunchPlanNotFound() {
+    doThrow(new StatusRuntimeException(Status.NOT_FOUND))
+        .when(flyteAdminClient).createExecution(any(), any(), any(), any(),any());
+
+    assertThrows(
+        FlyteRunner.LaunchPlanNotFound.class,
+        () -> flyteRunner.createExecution("exec", FLYTE_EXEC_CONF));
+  }
+
+  @Test
+  public void testThrowsCreateExecutionExceptionForOtherCode() {
+    doThrow(new StatusRuntimeException(Status.INTERNAL))
+        .when(flyteAdminClient).createExecution(any(), any(), any(), any(),any());
+
+    assertThrows(
+        FlyteRunner.CreateExecutionException.class,
+        () -> flyteRunner.createExecution("exec", FLYTE_EXEC_CONF));
+  }
+
+  @Test
+  public void testThrowsCreateExecutionExceptionForUnknownException() {
+    doThrow(new IllegalStateException("test"))
+        .when(flyteAdminClient).createExecution(any(), any(), any(), any(),any());
+
+    assertThrows(
+        FlyteRunner.CreateExecutionException.class,
+        () -> flyteRunner.createExecution("exec", FLYTE_EXEC_CONF));
+  }
+
+  @Test
+  public void testPollFlyteExecutionIdCannotBeNull() {
+    assertThrows(
+        NullPointerException.class,
+        () ->    flyteRunner.poll(null, RunState.create(null, null))
+    );
+  }
+
+  @Test
+  public void testPollRunStateCannotBeNull() {
+    assertThrows(
+        NullPointerException.class,
+        () ->    flyteRunner.poll(FlyteExecutionId.create("flyte-test", "testing", "test-run-state-cannot-be-null"), null)
+    );
+  }
+
+  @Test
+  public void testTransititionRunningToTerminateSuccessfulRun() throws Exception {
+    Workflow workflow = Workflow.create("id", configuration());
+    WorkflowInstance workflowInstance = WorkflowInstance.create(workflow.id(), "2016-03-14");
```

Comment:
These two lines appears verbatim 6 times in these file. I think that we should defined as constant

Reply:
i moved them in a function instead 👍